### DISCUSSION TITLE
How to deal with vue/multi-word-component-names ?

### DISCUSSION THREAD

**Discussion by laspinacristian** [posva/unplugin-vue-router#482] [Questions and Help] [answered]
When using this plugin, components under "pages" are imported created using the filename
However this naming strategy, while working for path in URLs, collide with vue3 recommended rule set for ESLint  

`Component name "index" should always be multi-word vue/multi-word-component-names`

Is there a recommended way to solve this issue?

Thank you

**Comment by laspinacristian** [comment, marked as answer]
ATM, I turned off the rule only for files inside `pages` folder only

If someone would stumble upon the same problem. this is my flat config file

```javascript
// @ts-check
import { readFileSync } from 'node:fs'
import pluginJs from '@eslint/js'
import tseslint from 'typescript-eslint'
import pluginVue from 'eslint-plugin-vue'
import eslintConfigPrettier from 'eslint-config-prettier'

const eslintAutoImport = JSON.parse(
  readFileSync('./tsconfig/eslint-auto-import.json', { encoding: 'utf8' })
)

export default [
  { languageOptions: eslintAutoImport },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  ...pluginVue.configs['flat/recommended'],
  {
    files: ['**/*.vue'],
    languageOptions: {
      parserOptions: {
        parser: tseslint.parser
      }
    }
  },
/* to solve unplugin-vue-router */
  {
    files: ['src/pages/**/*.vue'],
    rules: {
      'vue/multi-word-component-names': 'off' 
    }
  },
  eslintConfigPrettier
]

```

**Comment by oliverpool** [comment]
Based on https://stackoverflow.com/a/41316801/3207406, I created a `src/pages/.eslintrc` file with the following content:
```json
{
  "rules": {
    "vue/multi-word-component-names": "off"
  }
}
```

**Reply by posva** [reply]
That's probably for eslint 8 but it won't work for flat configs

**Reply by oliverpool** [reply]
You are right, this old config was generated upon vue project creation: https://github.com/vuejs/create-vue/issues/451

**Reply by ebisbe** [reply]
For anyone interested in a flat configuration you can use:

```
{
    files: ['src/pages/**/*.vue'],
    rules: {
      'vue/multi-word-component-names': 'off',
    },
  },

```
